test/CheckedC/type-checking-crash.c

```diff
@@ -0,0 +1,35 @@
+// Checks for crashes during type-checking error.
+// This was caused by an lvalue assert in SemaBounds.cpp
+// The goal of this test is to make sure we're seeing 
+// type-checking errors, but not get any crashes.
+// The llvm-lit test will fail the test if the compiler
+// crashes because of assert failure (This has been tested)
+// I wanted to use FileCheck to specifically look for

```

Comment:
The test looks good.    This comment needs to be cleaned up. I generally like tests to have comments with the following form:

- 1-2 sentence summary of what the test does.
- Blank comment line.
- More detailed explanation.

Comments should be written with the idea that someone you don't know will look at it 2 years from now.  The first sentence needs to be more precise.   I'd suggest removing the comment about trying to use FileCheck.  As you mention, llvm-lit will fail a test if a compiler crashes because of an assert in a compiler, so you can't use FileCheck to check if an assertion was hit.

It seems a little strange to me that we get a type checking error after we get a bounds checking error.   I think it would be clearer to users if the type checking message happened first (generally, you'd want to fix typechecking errors first).